### DISCUSSION TITLE
Welcome to kics Discussions!

### DISCUSSION THREAD

**Discussion by markmishaevcx** [Checkmarx/kics#737] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by subhamku2020** [comment]
Hi Team,

How do we scan a remote repository using checkmarx kics !

**Comment by rafaela-soares** [comment]
Hi @subhamku2020,

You need to use the prefix `git::`. For example: `docker run checkmarx/kics scan -p git::https://github.com/Checkmarx/kics`.

KICS provides other amazing ways of scanning, such as Terraformer (example: `docker run -e AWS_SECRET_ACCESS_KEY -e AWS_ACCESS_KEY_ID -e AWS_SESSION_TOKEN checkmarx/kics:latest scan -p "terraformer::aws:vpc:eu-west-2" -v --no-progress`) that scans resources deployed in the Cloud. 

It also provides Kuberneter (example: `docker run -v <credentials_path>:/credentials -v ${PWD}:/path/ -e K8S_CONFIG_FILE=/credentials/<config-file-name> checkmarx/kics:latest scan -p "kuberneter::*:*:*" -v --no-progress -o /path/results`) that scans resources deployed in the runtime K8s cluster. 

If you have interest, take a look at the documentation of [Terraformer](https://github.com/Checkmarx/kics/blob/master/docs/integrations_terraformer.md) and [Kuberneter](https://github.com/Checkmarx/kics/blob/master/docs/integrations_kuberneter.md).

You can also find other types of paths for the scan [here](https://docs.kics.io/latest/running-kics/).

Let me know if it helps you, please 😊

**Comment by subhamku2020** [comment]
Hi Rafaela Soares,

Thanks for your immediate response. Can you help me knowing if it creates a local copy of the repository before scanning and the additional parameter available along with prefix git to scan private repos.

**Comment by rafaela-soares** [comment]
Hi Subham Mishra (@subhamku2020), 

KICS uses `go-getter` to extract the git repos. Since KICS runs in Docker, the extracted repo folder will be in the temp folder of the container.

You can scan private repos by adding your [Personal Access Token](https://docs.github.com/pt/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token). For example: `docker run checkmarx/kics scan -p git::https://<personal-token-here>@github.com/<user>/<repo>.git`.

**Comment by subhamku2020** [comment]
Hi Rafaela Soares,

Thanks, for the response. Can you also let me know for solving issues w.r.t "SSL certificate problem: self signed certificate in certificate chain". It works fine to clone a project with git clone but not working with kics go-getter.

**Comment by rafaela-soares** [comment]
Hello @subhamku2020,

I did not come across this issue yet. I do not if it will be helpful but I found [this page](https://jhooq.com/2-ways-to-fix-ssl-certificate-problem-self-signed-certificate-in-certificate-chain/)
